src/shared/telemetry/awsTelemetryOptOut.ts

```diff
@@ -29,41 +28,42 @@ export class AwsTelemetryOptOut {
         private readonly getVSCodeTelemetrySetting: () => boolean = () =>
             !!vscode.workspace.getConfiguration('telemetry').get<boolean>('enableTelemetry')
     ) {
+        // TODO : update telemetry configuration when aws.telemetry changes (so that we can enable/disable without a restart)
         vscode.workspace.onDidChangeConfiguration(async event => {
             // if telemetry.enableTelemetry changed and user has not expressed a preference
             if (
                 event.affectsConfiguration('telemetry.enableTelemetry') &&
-                this.settings.readSetting(AwsTelemetryOptOut.AWS_TELEMETRY_KEY) === undefined
+                this.settings.readSetting(AwsTelemetryOptOut.AWS_TELEMETRY_KEY) === SETTINGS_TELEMETRY_VALUE_USEIDE
             ) {
-                await this.updateTelemetryConfiguration(TelemetryOptOutOptions.SameAsVsCode)
+                await this.updateTelemetryConfiguration(SETTINGS_TELEMETRY_VALUE_USEIDE)
             }
         })
     }
 
-    public async updateTelemetryConfiguration(response: TelemetryOptOutOptions) {
-        switch (response) {
-            case TelemetryOptOutOptions.Enable:
+    public async updateTelemetryConfiguration(telemetrySettingsValue: string) {
```

Comment:
Thoughts on this method `AwsFooOptOut.updateFooConfiguration(fooSettingsValue)`:

- Why do we prefix some things with "Aws" but not other things? All things in the project are AWS things.
- `AwsFooOptOut.updateFooConfiguration(fooSettingsValue)` repeats "foo" 3 times. Classes act as namespaces, but we've named the method as if we didn't have namespaces (like C).
- `fooSettingsValue` is a parameter in a `updateFooConfiguration` method, but we've named the parameter as if it exists in a vacuum. It's a parameter on the `updateFooConfiguration` method, so we know it's related to "Foo", we could omit that in the parameter name.
  - the phrase "settings value" itself is redundant: a setting is a value.

Is the repetition/redundancy helpful?

Reply:
Agreed -- I'll be working on a tidy up pass in this space as a separate PR and looking to sanitize some names.